presto-main/src/main/java/io/prestosql/connector/system/jdbc/ColumnJdbcTable.java

```diff
@@ -111,24 +139,155 @@ public ConnectorTableMetadata getTableMetadata()
         return METADATA;
     }
 
+    @Override
+    public TupleDomain<ColumnHandle> applyFilter(ConnectorSession connectorSession, Constraint constraint)
+    {
+        TupleDomain<ColumnHandle> tupleDomain = constraint.getSummary();
+        if (tupleDomain.isNone() || !constraint.predicate().isPresent()) {
+            return tupleDomain;
+        }
+        Predicate<Map<ColumnHandle, NullableValue>> predicate = constraint.predicate().get();
+        Set<ColumnHandle> predicateColumns = constraint.getColumns().orElseThrow(() -> new VerifyException("columns not present for a predicate"));
+
+        boolean hasSchemaPredicate = predicateColumns.contains(TABLE_SCHEMA_COLUMN);
+        boolean hasTablePredicate = predicateColumns.contains(TABLE_NAME_COLUMN);
+        if (!hasSchemaPredicate && !hasTablePredicate) {
+            // No filter on schema name and table name at all.
+            return tupleDomain;
+        }
+
+        Session session = ((FullConnectorSession) connectorSession).getSession();
+
+        Optional<String> catalogFilter = tryGetSingleVarcharValue(tupleDomain, TABLE_CATALOG_COLUMN);
+        Optional<String> schemaFilter = tryGetSingleVarcharValue(tupleDomain, TABLE_SCHEMA_COLUMN);
+        Optional<String> tableFilter = tryGetSingleVarcharValue(tupleDomain, TABLE_NAME_COLUMN);
+
+        if (catalogFilter.equals(Optional.of("")) ||
+                schemaFilter.equals(Optional.of("")) ||
+                tableFilter.equals(Optional.of(""))) {
+            return TupleDomain.none();
```

Comment:
It will be empty string when user query contains filters like `TABLE_CAT = ''`.
It is not really an optimization, because eg `SchemaTablePrefix` constrcuted below validates args are non-empty, so the code would fail without this `if` here.

Reply:
Will add a comment.

Reply:
This is same problem as in https://github.com/prestosql/presto/issues/2523. I will rework that fix, so that it's applicable to .jdbc.columns as well.